exec/java-exec/src/main/java/org/apache/drill/exec/store/PluginHandle.java

```diff
@@ -30,7 +31,7 @@
  * long Drillbit start times. Plugin creation is synchronized as is closing.
  * <p>
  * A handle has a type used to determine which operations are allowed on
- * the handle. For example, inrinsic (system) plugins cannot be deleted or
+ * the handle. For example, intrinsic (system) plugins cannot be deleted or
  * disabled.
  *
  * <h4>Caveats</h4>
```

Comment:
Please check javadoc for`PluginHandle`.
The plugin instances are not created separately for every user. Usually the instances are reused by different users (see an example for user1, user2, user3 in javadoc).
Therefore UserSession can't be kept with current plugins.
There is a solution In #2251 to have separate `StoragePluginRegistryImpl` for every session. That can be used for  per-user credentials then.

Reply:
The plan here is to actually use the credential provider to store different user's credentials.  I got this to work in a separate branch, but I was blocked because I didn't have the active user name.  This work here represents what is necessary just to get the active user name.